### ISSUE TITLE
Usage of `&alias` and `*alias` cause unintended "unexpected key"

### ISSUE THREAD

**Issue by buty4649** [smortex/yaml-sort#16]
Hi.

An error occurs when reading a YAML file with the correct format.

↓ test.yml
```yml
---
def: &alias
  test1: hoge
  test2: hige

zef:
  - def
  - abc
  - zxy

abc:
  <<: *alias
```

Using YAML.load_file does not result in an error.

```ruby
require 'yaml'
YAML.load_file("test.yml", aliases:true)
# => {"def"=>{"test1"=>"hoge", "test2"=>"hige"}, "zef"=>["def", "abc", "zxy"], "abc"=>{"test1"=>"hoge", "test2"=>"hige"}}
```

However, using yaml-sort results in an unexpected error.

```ruby
❯ yaml-sort test.yml
test.yml:6 unexpected KEY
zef:
^~~~
```

* ruby: `3.1.2p20`
* yaml-sort: `2.1.1`

**Comment by smortex** [comment]
Hi @buty4649 !

There is a "to do" item for this [in the project](https://github.com/smortex/yaml-sort/projects/1?query=is%3Aopen+sort%3Aupdated-desc).

The tricky part is that *I think* you have to move parts around as just sorting and keeping the aliases where they are will not work: the following cannot be parsed because `alias` is used before it is declared (raise `Unknown alias: alias (Psych::BadAlias)`):

```yaml
---
abc:
  <<: *alias
def: &alias
  test1: hoge
  test2: hige
zef:
  - def
  - abc
  - zxy
```

Your example could however be turned into this which would work:

```yaml
---
abc: &alias
  test1: hoge
  test2: hige
def:
  <<: *alias
zef:
  - def
  - abc
  - zxy
```

So this looks doable?  Yes but how to deal with the following:

```yaml
---
def: &alias
  test1: hoge
  test2: hige
abc:
  <<: *alias
  test3: foo
  test4: bar
```

I have no clue ATM, and my limited usage of aliases makes me think this is not something that can be reliably done. Any though?

**Comment by buty4649** [comment]
Thanks for the reply.

The problem was more difficult than I thought.
I think my problem can be solved with the workaround you gave me.

**Comment by smortex** [comment]
Maybe we can just ignore `<<: *alias` because it cannot be sorted. I opened #17 that sort anchors and aliases when they are in their "simple" form and raise an error otherwise.  Can you please give it a try and tell me if it helps in your use case?

**Comment by buty4649** [comment]
Thank you for #17 . I tried it and it worked fine in my use case.
If `<<: *alias` is included, I would consider replacing it up front.

e.g.
```yaml
❯ cat test.yml
---
abc: &alias
  test1: hoge
  test2: hige
def:
  <<: *alias
zef:
  - def
  - abc
  - zxy
❯ sed 's/<<: \*alias/ALIAS: __ALIAS__/g'  test.yml | tee temp.yml
---
abc: &alias
  test1: hoge
  test2: hige
def:
  ALIAS: __ALIAS__
zef:
  - def
  - abc
  - zxy
❯ bundle exec yaml-sort temp.yml | sed 's/ALIAS: __ALIAS__/<<: *alias/g'
---
abc: &alias
  test1: hoge
  test2: hige
def:
  <<: *alias
zef:
  - def
  - abc
  - zxy
```

**Comment by smortex** [comment]
Thank your for your feedback. I released v2.2.0 a few minutes ago with support for one-line anchors and aliases.
